src/main.qml

```diff
@@ -129,53 +156,268 @@ Application {
                 }
 
-                Icon {
-                    name: IconTools.getIconName(owmId.value)
-                    anchors.centerIn: parent
-                    width: Dims.w(33)
+                Rectangle {
+                    id: iconCircle
+
+                    width: Dims.w(24)
                     height: width
+                    radius: width/2
+                    color: "#00ffffff"
+                    anchors {
+                        centerIn: parent
+                        horizontalCenterOffset: -Dims.w(30)
+                    }
+
+                    Icon {
+                        id: iconDisplay
+
+                        name: IconTools.getIconName(owmId.value)
+                        width: Dims.w(24)
+                        height: width
+                        opacity: 1
+                        anchors.centerIn: parent
+                    }
                 }
 
-                Label {
-                    //% "Max:"
-                    text: "<h6>" + qsTrId("id-max") + "</h6>\n" + convertTemp(maxTemp.value)
-                    horizontalAlignment: Text.AlignHCenter
-                    verticalAlignment: Text.AlignVCenter
-                    anchors.bottom: parent.bottom
-                    anchors.top: parent.top
-                    anchors.right: parent.right
-                    width: Dims.w(33)
-                    font.pixelSize: Dims.l(6)
+                Text {
+                    id: tempBig
+
+                    property string tempLength: convertTempRaw(Math.round((maxTemp.value + minTemp.value) / 2))
+                    property string tempSliceMinus: tempLength.slice(0, 1) === "-" ? tempLength.slice(1, 4) : tempLength
+
+                    color: "#ffffffff"
+                    renderType: Text.NativeRendering
+                    anchors {
+                        verticalCenter: parent.verticalCenter
+                        verticalCenterOffset: -Dims.h(5)
+                        horizontalCenter: parent.horizontalCenter
+                    }
+                    lineHeight: Dims.l(0.2)
+                    font {
+                        letterSpacing: -Dims.l(1)
+                        styleName: "ExtraCondensed Thin"
+                        pixelSize: Dims.w(34)
+                    }
+                    text: tempSliceMinus
+
+                    Text {
+                        color: "#ffffffff"
+                        renderType: Text.NativeRendering
+                        anchors {
+                            left: tempBig.right
+                            leftMargin: Dims.w(0.4)
+                            top: tempBig.top
+                            topMargin: Dims.h(7.0)
+                        }
+                        font {
+                            styleName: "Light"
+                            pixelSize: Dims.w(15)
+                        }
+                        text: "°"
+                    }
+
+                    Text {
```

Comment:
Using Text is kinda ok in watchfaces where you want raw control over the text rendering, but apps are meant to use Labels instead.
A Label is essentially just a Text with a set of default properties that make sense for AsteroidOS https://github.com/AsteroidOS/qml-asteroid/blob/master/src/controls/qml/Label.qml#L21
Most importantly, it sets the default color to white.
If you want to override the default you can keep the exact same code, but you should be able to inherit some of the defaults and keep the code simpler

Reply:
Yepp, the color being fixed to white is what made me choose Text.
I now realize i could have asked whats the purpose od Label at all.

Reply:
Had some trouble. 
I was thinking i can not align to .right .left .bottom and .top anchors since labels do not have them.
Turns out i was wrong and @MagneFire solved the issue by not letting bigTemp clip. thx again!